### PR TITLE
Add a way to find the country of a zone

### PR DESCRIPTION
**Pull request by antoine-de**
it's a bit of a 🥚 🐔 problem, we need to know the country of a zone to find the libpostal's rules to find the `ZoneType` and we need the libpostal rules to know which `admin_level` is a country (it's not always `2`...)

So we use OSM's tag "ISO3166-1:alpha2" to know if a zone is a country

Note: this PR depends on #21 and should be merged after it. it might be easier to review [with the diff](https://github.com/amatissart/cosmogony/compare/zonesContainment...antoine-de:find_country)

### CODE REVIEW

File: src/country_finder.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                    let mut candidates: Vec<_> = self.tree
+                        .get(&bbox_to_rect(bbox))
+                        .into_iter()
+                        .map(|(_, country)| country)
+                        .collect();
+
+                    candidates.sort_by_key(|c| -1 * c.zone.admin_level.unwrap_or(0) as i32);
+                    candidates
+                        .iter()
+                        .filter(|c| c.ggeom.contains(&(&z.boundary.clone().unwrap()).into()))
```
Comment (amatissart) [inline src/country_finder.rs]: I'm reviewing my own code here, but that should definitely be simplified.  
Actually, explicit `GGeom` is probably not even useful here.

File: src/lib.rs [hunk 2]
```diff
@@ -87,46 +98,80 @@ pub fn get_zones_and_stats_without_geom(
...
 
     for mut z in zones {
-        let country = get_country(&z, &country_code)?;
-        let type_res = zone_typer.get_zone_type(&z, &country);
-        match type_res {
-            Ok(t) => z.zone_type = Some(t),
-            Err(zone_typer::ZoneTyperError::InvalidCountry(c)) => {
-                info!("impossible to find {}", c);
-                *stats.zone_with_unkwown_country.entry(c).or_insert(0) += 1;
+        let country_code = get_country_code(&country_finder, &z, &country_code);
```
Comment (amatissart) [inline src/lib.rs]: I fear that there might be an uppercase / lowercase difference between the code returned by the country_finder and the libpostal filenames. Are we sure that's handled correctly ?

Comment (antoine-de) [reply src/lib.rs]: it is done while reading the countrycode, but you're right I moved it in the libpostal lookup, it's more consistent

File: Cargo.lock [hunk 3]
```diff
@@ -88,13 +88,13 @@ dependencies = [
 [[package]]
 name = "bragi"
 version = "1.2.0"
-source = "git+https://github.com/CanalTP/mimirsbrunn.git#474972c1b0547d99bea8e8c74e10c499f81c10b5"
+source = "git+https://github.com/antoine-de/mimirsbrunn.git?branch=fix_lat_lon#6738b8171cdddaa68702852c28572a74bf6dd0a1"
```
Comment (amatissart) [inline Cargo.lock]: Your repo should no longer be referenced in `Cargo.lock`

File: src/hierarchy_builder.rs [hunk 4]
```diff
@@ -46,16 +46,6 @@ impl ZonesTree {
...
-    fn from_iter<I: IntoIterator<Item = &'a Zone>>(zones: I) -> Self {
-        let mut ztree = ZonesTree::default();
-        for z in zones {
-            ztree.insert_zone(z);
-        }
-        ztree
-    }
-}
-
 pub fn bbox_to_rect(bbox: Bbox<f64>) -> Rect {
```
Comment (amatissart) [inline src/hierarchy_builder.rs]: This method is now defined in utils.rs and should be removed.

Comment (amatissart) [reply src/hierarchy_builder.rs]: The same for `down` and `up` methods, later in file.

**Pull request merged**